### ISSUE TITLE
Windows Support

### ISSUE THREAD

**Issue by jc324** [jerel/membrane#8]
Hi,

Cool library. I would like to suggest windows support, that would be really helpful. So far, I'm just manually patching `loader.dart` with this:

```dart
if (Platform.isWindows) {
  Logger('membrane').info('Opening native library example.dll');
  return DynamicLibrary.open('example.dll');
}
```

Thanks.

**Comment by jc324** [comment]
I'm not sure how membrane automatically moves the shared libs to the root of the dart package, but it'd also be great to get the Windows DLLs moved.

**Comment by LambdaMan2K** [comment]
And I just tried the example on windows 10 x64 and got errors like this

```
#[async_dart(namespace = "locations")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i64`, 
found `i32`
```

**Comment by jerel** [comment]
The moving of the so/dylib/dll is usually done via something like xtask. Membrane doesn't automatically do the moving for you so you can choose how you want it done as part of your build process.

I don't have access to a Windows machine to develop on but if it mostly Just Works<sup>:tm:</sup> and needs the loader addition then I'm happy to accept that. Would you want to open a pull request?

So @jc324 is it working for you on Windows but not working at all for @LambdaMan2K? Is there a difference in OS versions or something?

**Comment by LambdaMan2K** [comment]
well I just ried the example and  it wont compile do to wrong input somehow.  it seems strange if this runs for others :)

**Comment by jc324** [comment]
I'm running it on Windows 11, maybe that makes a difference, idk. In `example/src/application.rs` these two functions work:
```rust
#[async_dart(namespace = "accounts")]
pub fn contacts() -> impl Stream<Item = Result<data::Contact, data::Error>> {
  futures::stream::iter(vec![Ok(data::Contact::default())])
}

#[async_dart(namespace = "accounts")]
pub async fn contact(user_id: String) -> Result<data::Contact, data::Error> {
  Ok(data::Contact {
    id: user_id.parse().unwrap(),
    ..data::Contact::default()
  })
}
```

Most of the rest fail with the error mentioned by @LambdaMan2K.

> ```
> #[async_dart(namespace = "locations")]
>     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i64`, 
> found `i32`
> ```

There also seems to be an error on the dart side when calling a rust function that accepts `Vec<String>` as an arg.

**Comment by jc324** [comment]
Yes, I got those errors too. But using the gnu tool chain worked fine for
me, at least for sending basic data over.

**Comment by jerel** [comment]
Closing this since the windows loader was added in a PR
